test/widgets/test_window_count.py

```diff
@@ -1,10 +1,66 @@
+import pytest
+
 import libqtile
+from libqtile.confreader import Config
 from libqtile.widget import WindowCount
 
 
+class DifferentScreens(Config):
+    groups = [
+        libqtile.config.Group("a"),
+        libqtile.config.Group("b"),
+    ]
+    layouts = [
+        libqtile.layout.Stack(num_stacks=1),
+    ]
+    floating_layout = libqtile.resources.default_config.floating_layout
+    fake_screens = [
+        libqtile.config.Screen(
+            top=libqtile.bar.Bar(
+                [
+                    WindowCount(),
+                ],
+                20
+            ),
+            x=0, y=0, width=300, height=300
+        ),
+        libqtile.config.Screen(
+            top=libqtile.bar.Bar(
+                [
+                    WindowCount(),
+                ],
+                20
+            ),
+            x=0, y=300, width=300, height=300
+        )
+    ]
+    auto_fullscreen = True
+
+
+different_screens = pytest.mark.parametrize("manager", [DifferentScreens], indirect=True)
+
+
+@different_screens
+def test_different_screens(manager):
+    # Put one window on screen 0
+    manager.c.to_screen(0)
+    manager.test_window("one")
+
+    # Put two windows on screen 1
+    manager.c.to_screen(1)
+    manager.test_window("two")
+    manager.test_window("three")
+
+    assert manager.c.screen[0].widget["windowcount"].get() == "1"
+    assert manager.c.screen[1].widget["windowcount"].get() == "2"
+
+
 def test_window_count(manager_nospawn, minimal_conf_noscreen):
     config = minimal_conf_noscreen
     config.screens = [
+            libqtile.config.Screen(
+                top=libqtile.bar.Bar([WindowCount()], 10)
+            ),
```

Comment:
Don't think we need these 3 lines. Unless I'm missing something.

Reply:
Oh, yes I added them when I was trying to write tests and forgot them.
I removed them.